### PR TITLE
Fixed exception message

### PR DESCRIPTION
**Pull request by svycka**

### CODE REVIEW

File: src/ServiceManager.php [hunk 1]
```diff
@@ -276,9 +276,10 @@ public function setFactory($name, $factory, $shared = null)
         $cName = $this->canonicalizeName($name);
 
         if (!($factory instanceof FactoryInterface || is_string($factory) || is_callable($factory))) {
-            throw new Exception\InvalidArgumentException(
-                'Provided abstract factory must be the class name of an abstract factory or an instance of an AbstractFactoryInterface.'
-            );
+            throw new Exception\InvalidArgumentException(sprintf(
+                'Provided abstract factory must be the class name of an factory or an instance of an "%s".',
```
Comment (Maks3w) [inline src/ServiceManager.php]: Missed callable is allowed too

**Comment by svycka** [comment]
@Maks3w thanks fixed

File: src/ServiceManager.php [hunk 2]
```diff
@@ -276,9 +276,10 @@ public function setFactory($name, $factory, $shared = null)
         $cName = $this->canonicalizeName($name);
 
         if (!($factory instanceof FactoryInterface || is_string($factory) || is_callable($factory))) {
-            throw new Exception\InvalidArgumentException(
-                'Provided abstract factory must be the class name of an abstract factory or an instance of an AbstractFactoryInterface.'
-            );
+            throw new Exception\InvalidArgumentException(sprintf(
+                'Provided factory must be the class name of an factory, callable or an instance of an "%s".',
```
Comment (danizord) [inline src/ServiceManager.php]: `an` -> `a`

Comment (samsonasik) [inline src/ServiceManager.php]: also, `an instance of an "%s"` -> `an instance of "%s"`

**Pull request merged**